### DISCUSSION TITLE
rotational grazing

### DISCUSSION THREAD

**Discussion by KeLiu7** [APSIMInitiative/ApsimX#9104] [Q&A] [answered]
Dear all,

Do you know how to set rotational grazing with different rest periods in APSIM Next Gen? For example, grazing on 31 Jan then rest for five month, then grazing on 31 Jul and rest for five months (ideally it should be like Fig.1 ) but it was the Fig. 2. 

![image](https://github.com/APSIMInitiative/ApsimX/assets/72141690/f095c80b-bf68-4eee-a524-c1a144e7cdf4)
![image](https://github.com/APSIMInitiative/ApsimX/assets/72141690/db9b0d16-c9f3-46b8-9640-e6260379062f)

I have tried to “GrazeForageCrop” to set this, but my scripts do not work as I expected. Could you please have a look when you available? Many thanks😊
![image](https://github.com/APSIMInitiative/ApsimX/assets/72141690/a134f602-41b3-4c20-a618-16c60148f6a3)
[rotational grazing examples.zip](https://github.com/user-attachments/files/16109928/rotational.grazing.examples.zip)

**Comment by sno036** [comment, marked as answer]
When you added more no-grazing windows to the script you left the lines

```
             if (DoNoGrazingWindow)
                GrazingSeasonIsOpen = !DateUtilities.WithinDates(NoGrazeStart, clock.Today, NoGrazeEnd);
```

untouched so they only look at the first set of dates.

If you only want the crop grazed on two days of the year why not use Operations?

**Reply by KeLiu7** [reply]
Many thanks Val for your explanation. Now it works as I expected😊
